### PR TITLE
S3N extension project

### PR DESCRIPTION
**Pull request by madanadit**
- alluxio-core-common w/ scope: provided
- remove shading
- add dependency versions

### CODE REVIEW

**Comment by madanadit** [comment]
@apc999 @aaudiber

File: underfs/s3n/pom.xml [hunk 1]
```diff
@@ -12,55 +12,134 @@
...
-    <!-- The following paths need to be defined here as well as in the parent pom so that mvn can -->
-    <!-- run properly from sub-project directories -->
-    <license.header.path>${project.parent.parent.basedir}/build/license/</license.header.path>
-    <checkstyle.path>${project.parent.parent.basedir}/build/checkstyle/</checkstyle.path>
-    <findbugs.path>${project.parent.parent.basedir}/build/findbugs/</findbugs.path>
-    <!-- The shading prefix should match the artifact ID, replacing '-' with '.' -->
-    <shading.prefix>alluxio.underfs.s3</shading.prefix>
+    <build.path>build</build.path>
+    <checkstyle.path>build/checkstyle/</checkstyle.path>
+    <findbugs.path>build/findbugs</findbugs.path>
```
Comment (apc999) [inline underfs/s3n/pom.xml]: there is no build/findbugs in this repo, do we want to copy them here?

File: underfs/s3n/pom.xml [hunk 2]
```diff
@@ -12,55 +12,134 @@
...
-      <version>${project.version}</version>
     </dependency>
 
     <!-- External test dependencies -->
     <dependency>
       <groupId>org.alluxio</groupId>
       <artifactId>alluxio-core-common</artifactId>
-      <version>${project.version}</version>
       <type>test-jar</type>
-      <scope>test</scope>
```
Comment (apc999) [inline underfs/s3n/pom.xml]: why move `<scope>test</scope>` to dependency management?
I would suggest them stay here

File: underfs/s3n/pom.xml [hunk 3]
```diff
@@ -121,54 +216,10 @@
...
           </execution>
         </executions>
       </plugin>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
-          <testExcludes>
-            <!-- Skip UFS contract tests unless -DtestS3Bucket="s3n://my-bucket/alluxio-test" is specified -->
-            <exclude>**/S3UnderFileSystemContractTest.java</exclude>
```
Comment (apc999) [inline underfs/s3n/pom.xml]: if this part removed, the mvn test will fail

File: underfs/s3n/pom.xml [hunk 4]
```diff
@@ -12,55 +12,134 @@
...
+      <dependency>
+        <groupId>org.alluxio</groupId>
+        <artifactId>alluxio-core-common</artifactId>
+        <version>${project.version}</version>
+        <scope>provided</scope>
+      </dependency>
+
+      <!-- External test dependencies -->
+      <dependency>
+        <groupId>org.alluxio</groupId>
```
Comment (apc999) [inline underfs/s3n/pom.xml]: sort by groupId then artifactId?

File: underfs/s3n/pom.xml (hunk 3 above)
Comment (apc999) [inline underfs/s3n/pom.xml]: ignore my comment here

File: underfs/s3n/pom.xml (hunk 2 above)
Comment (madanadit) [inline underfs/s3n/pom.xml]: is there a benefit to redefining the scope here when it is already defined in dependencyManagement?

File: underfs/s3n/pom.xml (hunk 1 above)
Comment (madanadit) [inline underfs/s3n/pom.xml]: copied it

**Comment by aaudiber** [comment]
Should we rename the maven artifact to s3n instead of just s3?

I'd like to make the pom and build infrastructure as minimal as possible so that others can use this repo as a model for creating new connectors. 

@madanadit @apc999 Do you guys think findbugs/checkstyle are necessary? Now that we are using an isolated classloader for loading extensions, do we still need shading?

**Comment by madanadit** [comment]
The shading plugin is used just for assembly (creating the uber jar), i.e., there is no relocation of packages.

About findbugs/checkstyle, I'm fine either way. I can remove it if Bin is okay.

**Comment by madanadit** [comment]
@apc999 are you okay w/ removing findbugs/checkstyle?

File: underfs/s3n/pom.xml [hunk 5]
```diff
@@ -12,54 +12,136 @@
...
+      <!-- External test dependencies -->
+      <dependency>
+        <groupId>junit</groupId>
+        <artifactId>junit</artifactId>
+        <version>${junit.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.alluxio</groupId>
+        <artifactId>alluxio-core-common</artifactId>
```
Comment (apc999) [inline underfs/s3n/pom.xml]: this should be in a session of `<!-- Core Alluxio test dependencies -->`

File: underfs/s3n/pom.xml [hunk 6]
```diff
@@ -12,54 +12,136 @@
...
+        <artifactId>jets3t</artifactId>
+        <version>${jets3t.version}</version>
+      </dependency>
+
+      <!-- Core Alluxio dependencies -->
+      <dependency>
+        <groupId>org.alluxio</groupId>
+        <artifactId>alluxio-core-common</artifactId>
+        <version>${project.version}</version>
+        <scope>provided</scope>
```
Comment (apc999) [inline underfs/s3n/pom.xml]: don't need scope in dependency management

File: underfs/s3n/pom.xml [hunk 7]
```diff
@@ -12,54 +12,136 @@
...
+      </dependency>
+
+      <!-- External test dependencies -->
+      <dependency>
+        <groupId>junit</groupId>
+        <artifactId>junit</artifactId>
+        <version>${junit.version}</version>
+        <scope>test</scope>
```
Comment (apc999) [inline underfs/s3n/pom.xml]: don't need scope in dependency management

**Comment by madanadit** [comment]
@aaudiber Please take another look.

**Pull request merged**